storagemarket/deal_commp.go

```diff
@@ -196,7 +196,8 @@ func getCarSize(filepath string, rd *carv2.Reader) (int64, error) {
 	var size int64
 	switch rd.Version {
 	case 2:
-		size = int64(rd.Header.DataSize)
+		padding := rd.Header.DataOffset - carv2.PragmaSize - carv2.HeaderSize
+		size = int64(rd.Header.DataSize) + int64(padding)
```

Comment:
This is certainly changing the `size`, which is then used in `storagemarket/deal_commp.go:131`. I am a bit confused how that is not changing the behavior and breaking existing functionality...

Reply:
My guess is we haven't had any V2 CAR files that had any padding.